smtpmail/backend/plugin/src/main/kotlin/com/ritense/valtimo/backend/plugin/plugin/SmtpMailPlugin.kt

```diff
@@ -0,0 +1,113 @@
+import com.ritense.valueresolver.ValueResolverService
+import org.camunda.bpm.engine.delegate.DelegateExecution
+
+@Plugin(
+    key = "smtp-mail",
+    title = "SMTP mail Plugin",
+    description = "Send mail through SMTP with the SMTP mail plugin"
+)
+class SmtpMailPlugin(
+    private val smtpMailService: SmtpMailService,
+    private val valueResolverService: ValueResolverService
+) {
+
+    @PluginProperty(key = "host", secret = false, required = true)
+    lateinit var host: String
+
+    @PluginProperty(key = "port", secret = false, required = true)
+    lateinit var port: String
+
+    @PluginProperty(key = "username", required = false, secret = false)
+    var username: String? = null
+
+    @PluginProperty(key = "password", required = false, secret = true)
+    var password: String? = null
+
+    @PluginProperty(key = "protocol", required = false, secret = false)
+    var protocol: String? = "smtp"
+
+    @PluginProperty(key = "debug", required = false, secret = false)
+    var debug: Boolean? = true
+
+    @PluginProperty(key = "auth", required = false, secret = false)
+    var auth: Boolean? = true
+
+    @PluginProperty(key = "startTlsEnable", required = false, secret = false)
+    var startTlsEnable: Boolean? = true
+
+    @PluginAction(
+        key = "send-mail",
+        title = "Send mail",
+        description = "Send an email",
+        activityTypes = [ActivityType.SERVICE_TASK_START]
+    )
+
+    @Suppress("UNCHECKED_CAST")
+    fun sendMail(
+        execution: DelegateExecution,
+        @PluginActionProperty sender: String,
+        @PluginActionProperty recipients: String,
+        @PluginActionProperty cc: String?,
+        @PluginActionProperty bcc: String?,
+        @PluginActionProperty subject: String,
+        @PluginActionProperty contentId: String,
+        @PluginActionProperty attachmentIds: String?,
+    ) {
+        val recipientList: List<String> = (resolveValue(execution, recipients)) as List<String>? ?: emptyList()
+        val ccList: List<String> = (resolveValue(execution, cc)) as List<String>? ?: emptyList()
+        val bccList: List<String> = (resolveValue(execution, bcc)) as List<String>? ?: emptyList()
+        val attachmentIdList: List<String> = (resolveValue(execution, attachmentIds)) as List<String>? ?: emptyList()
+
+        smtpMailService.sendSmtpMail(
+            mailContext = SmtpMailContextDto(
+                sender = Email(resolveValue(execution, sender) as String),
+                recipients = recipientList.map { Email (it) },
+                ccList = ccList.map { Email(it) },
+                bccList = bccList.map { Email(it) },
+                subject = resolveValue(execution, subject) as String,
+                contentResourceId = resolveValue(execution, contentId) as String,
+                attachmentResourceIds = attachmentIdList
+            )
+        )
+    }
+
+    private fun resolveValue(execution: DelegateExecution, value: String?): Any? {
```

Comment:
I guess this is a work-around for the bug. If it works, it works

Reply:
actually, we wanted to check if this was still necessary, but because of the bug I cant :)